test/e2e-tests.sh

```diff
@@ -36,23 +42,33 @@ function knative_setup() {
   # Install the latest Knative/eventing in the current cluster
   echo ">> Starting Knative Eventing"
   echo "Installing Knative Eventing"
-  ko apply -f config/ || return 1
+  ko apply -f ${EVENTING_CONFIG} || return 1
   wait_until_pods_running knative-eventing || fail_test "Knative Eventing did not come up"
 
   echo "Installing In-Memory ClusterChannelProvisioner"
-  ko apply -f config/provisioners/in-memory-channel/in-memory-channel.yaml || return 1
+  ko apply -f ${IN_MEMORY_CHANNEL_CONFIG} || return 1
   wait_until_pods_running knative-eventing || fail_test "Failed to install the In-Memory ClusterChannelProvisioner"
+
+  echo "Installing GCPPubSub ClusterChannelProvisioner"
+  kubectl -n knative-eventing create secret generic gcppubsub-channel-key --from-file=key.json=${GOOGLE_APPLICATION_CREDENTIALS}
+  E2E_CLUSTER_PROJECT="$(gcloud config get-value project)"
+  readonly E2E_CLUSTER_PROJECT
+  sed "s/REPLACE_WITH_GCP_PROJECT/${E2E_CLUSTER_PROJECT}/" ${GCP_PUBSUB_CHANNEL_CONFIG} | ko apply -f -
+  wait_until_pods_running knative-eventing || fail_test "Failed to install the GCPPubSub ClusterChannelProvisioner"
 }
 
 function knative_teardown() {
   echo ">> Stopping Knative Eventing"
   echo "Uninstalling Knative Eventing"
-  ko delete --ignore-not-found=true --now --timeout 60s -f config/
+  ko delete --ignore-not-found=true --now --timeout 60s -f ${EVENTING_CONFIG}
 
-  wait_until_object_does_not_exist namespaces knative-eventing
+  echo "Uninstalling In-Memory ClusterChannelProvisioner"
+  ko delete --ignore-not-found=true --now --timeout 60s -f ${IN_MEMORY_CHANNEL_CONFIG}
+
+  echo "Uninstalling GCPPubSub ClusterChannelProvisioner"
+  sed "s/REPLACE_WITH_GCP_PROJECT/${E2E_CLUSTER_PROJECT}/" ${GCP_PUBSUB_CHANNEL_CONFIG} | ko delete --ignore-not-found=true --now --timeout 60s -f -
```

Comment:
Instead of doing patches on the fly, create a temp file with the changes, then use it in setup and teardown. This will allow future more complex patches, while also keeping the final YAML file around for local debugging.

Reply:
Done.

Reply:
Unresolving to keep track since the code wasn't pushed yet.